vcd/resource_vcd_nsxv_firewall_rule.go

```diff
@@ -0,0 +1,850 @@
+package vcd
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"text/tabwriter"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdNsxvFirewall() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdNsxvFirewallCreate,
+		Read:   resourceVcdNsxvFirewallRead,
+		Update: resourceVcdNsxvFirewallUpdate,
+		Delete: resourceVcdNsxvFirewallDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdNsxvFirewallImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which Firewall Rule is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Firewall rule name",
+			},
+			"above_rule_id": &schema.Schema{
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+				Description: "This firewall rule will be inserted above the referred one",
+			},
+			"rule_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Computed:    true,
+				Description: "Read only. Possible values 'user', 'internal_high'",
+			},
+			"rule_tag": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
```

Comment:
`ForceNew:    true,` ?

Reply:
Yes, because the tag is so odd that it should alter the ID.

Reply:
Could you add some comment, as who haven't developed this resource wouldn't know why